.circleci/config.yml

```diff
@@ -172,6 +172,51 @@ jobs:
             docker login -u "$DOCKER_USER" -p "$DOCKER_PASS"
             make -C ./docker version=${CIRCLE_TAG} alias=latest push
 
+  publish-docker-image-arm64:
+    machine:
+      image: ubuntu-2004:current
+    resource_class: arm.medium
+    steps:
+      - checkout
+      - run:
+          name: check if emscripten-releases-tags.json has changed
+          command: |
+            CHANGED_FILES=$(git diff --name-only <<pipeline.git.base_revision>>..<<pipeline.git.revision>>)
+            if [[ ! $CHANGED_FILES =~ "emscripten-releases-tags.json" ]]; then
+              echo "emscripten-releases-tags.json has not changed, skipping build and publish"
+              exit 0
+            fi
+      - run:
+          name: extract latest emscripten arm64 linux version
+          command: |
+            export ARM64_VERSION=$(grep -oP '(?<="latest-arm64-linux": ")[^"]*' emscripten-releases-tags.json)
+            export ARM64_IMAGE_TAG=${ARM64_VERSION}-arm64
+            echo "ARM64_VERSION=$ARM64_VERSION"
+            echo "ARM64_IMAGE_TAG=$ARM64_IMAGE_TAG"
+      - run:
+          name: check if docker image already exists
+          command: |
+            DOCKER_IMAGE_EXISTS=$(curl -s -o /dev/null -w "%{http_code}" https://hub.docker.com/v2/repositories/emscripten/emsdk/tags/$ARM64_IMAGE_TAG)
+            if [ "$DOCKER_IMAGE_EXISTS" = "200" ]; then
+              echo "Docker image already exists, skipping build and publish"
+              exit 0
+            fi
+      - run:
+          name: install docker
+          command: apt-get update -q && apt-get install -q -y docker.io
+      - setup_remote_docker
+      - run:
+          name: build
+          command: make -C ./docker version=${ARM64_VERSION} build
+      - run:
+          name: test
+          command: make -C ./docker version=${ARM64_VERSION} test
+      - run:
+          name: push image
+          command: |
+            docker login -u "$DOCKER_USER" -p "$DOCKER_PASS"
+            make -C ./docker version=${ARM64_VERSION} alias=${ARM64_IMAGE_TAG} only_alias=true push
```

Comment:
What would the preferred tag be for these images? This PR would currently set it to `<emscripten-version>-arm64` e.g. `3.1.33-arm64`

Reply:
Does docker have a convention for this?     Is the architecture not an inherent part of the image?    What happens if you try to install the normal x64 imagine and arm host?

Reply:
Great questions and thanks for prompt reply!

> Does docker have a convention for this? Is the architecture not an inherent part of the image?

Docker does offer an experimental solution with [Docker Manifest](https://docs.docker.com/engine/reference/commandline/manifest/).

[The official docker guide](https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/) would suggest tagging your images with an arch suffix e.g. build each arch image separately and  tag with `3.1.33-amd64` and `3.1.33-arm64v8`. Then with the `docker manifest` tool you can alias these tags under `3.1.33` and push to the docker repo. When a consumer pulls an image like `emscripten/emsdk:3.1.33`, docker can auto select the appropriate arch for the host machine.

Moving to the manifest solution will increase the complexity of the current CI docker steps. Perhaps something that could be followed up with in the future?

> What happens if you try to install the normal x64 image and arm host?

Let me know if I've misunderstood. If a user installed both an `amd64` and `arm64v8` images they would have different docker tags so there's no conflict. The users would have to explicitly opt-in to using ARM based images, unless the manifest solution above was implemented.